core/src/test/java/com/exadel/aem/toolkit/core/injectors/DelegateInjector.java

```diff
@@ -62,39 +68,89 @@ public Object getValue(
         String effectiveName = StringUtils.defaultIfBlank(requestProperty.name(), name);
 
         if (delegate instanceof RequestAttributeInjector) {
-            return ((RequestAttributeInjector) delegate).getValue(
+            Object value = ((RequestAttributeInjector) delegate).getValue(
                 AdaptationUtil.getRequest(adaptable),
                 effectiveName,
                 type);
+            boolean isFallbackValue = value instanceof CastResult && ((CastResult) value).isFallback();
+            if ((value == null || isFallbackValue) && annotatedElement.isAnnotationPresent(Default.class)) {
+                value = getDefaultValue(annotatedElement.getDeclaredAnnotation(Default.class));
+                value = CastUtil.toType(value, type);
+            }
+            return value instanceof CastResult ? ((CastResult) value).getValue() : value;
         }
 
         if (delegate instanceof RequestParamInjector) {
-            return ((RequestParamInjector) delegate).getValue(
+            Object value = ((RequestParamInjector) delegate).getValue(
                 AdaptationUtil.getRequest(adaptable),
                 effectiveName,
                 type);
+            boolean isFallbackValue = value instanceof CastResult && ((CastResult) value).isFallback();
```

Comment:
I'm afraid the same five lines of code are copy-pasted five times. Suboptimal

Reply:
fixed